src/devices/ServoMotor/ServoMotor.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Pwm;
+
+namespace Iot.Device.ServoMotor
+{
+    /// <summary>
+    /// Represents a rotary actuator or linear actuator that allows for precise control of angular or linear position.
+    /// </summary>
+    public class ServoMotor : IDisposable
+    {
+        private PwmChannel _pwmChannel;
+        private readonly int _maximumAngle;
+        private readonly int _maximumPulseWidthMicroseconds;
+        private readonly int _minimumPulseWidthMicroseconds;
+        private readonly double _angleInMicroseconds;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServoMotor"/> class.
+        /// </summary>
+        /// <param name="maximumAngle">The maximum angle the servo motor can move.</param>
+        /// <param name="minimumPulseWidthMicroseconds">The minimum pulse width, in microseconds, that represent an angle for 0 degrees.</param>
+        /// <param name="maximumPulseWidthMicroseconds">The maxnimum pulse width, in microseconds, that represent an angle for maximum angle.</param>
+        private ServoMotor(int maximumAngle, int minimumPulseWidthMicroseconds, int maximumPulseWidthMicroseconds)
+        {
+            _maximumAngle = maximumAngle;
+            _minimumPulseWidthMicroseconds = minimumPulseWidthMicroseconds;
+            _maximumPulseWidthMicroseconds = maximumPulseWidthMicroseconds;
+            _angleInMicroseconds = CalculateAngleToMicroseconds(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds);
+        }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="ServoMotor"/> class.
+        /// </summary>
+        /// <param name="pwmChannel">The PWM channel used to control the servo motor.</param>
+        /// <param name="maximumAngle">The maximum angle the servo motor can move.</param>
+        /// <param name="minimumPulseWidthMicroseconds">The minimum pulse width, in microseconds, that represent an angle for 0 degrees.</param>
+        /// <param name="maximumPulseWidthMicroseconds">The maxnimum pulse width, in microseconds, that represent an angle for maximum angle.</param>
+        public ServoMotor(
+            PwmChannel pwmChannel,
+            int maximumAngle = 180,
+            int minimumPulseWidthMicroseconds = 1_000,
+            int maximumPulseWidthMicroseconds = 2_000)
+            : this(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds)
+        {
+            _pwmChannel = pwmChannel;
+        }
+
+        private static double CalculateAngleToMicroseconds(int maximumAngle, int minimumPulseWidthMicroseconds, int maximumPulseWidthMicroseconds)
+        {
+            if (maximumAngle < 0 || maximumAngle > 360)
```

Comment:
NIT: I know that this method is only called once during construction, so this might be a micro optimization that is not worth it, but you could perform both of this checks in one operation if you did: `if (((uint)maximumAngle) > 360)` basically by casting you would be making maximumAngle to be a very large number so it would be caught by the check if the number is > 360.

Reply:
Will update in next commit